.github/shared/src/swagger.js

```diff
@@ -2,16 +2,30 @@
 
 import $RefParser, { ResolverError } from "@apidevtools/json-schema-ref-parser";
 import { readFile } from "fs/promises";
-import { dirname, relative, resolve } from "path";
+import { basename, dirname, relative, resolve } from "path";
 import { mapAsync } from "./array.js";
 import { includesFolder } from "./path.js";
 import { SpecModelError } from "./spec-model-error.js";
+import { existsSync, readFileSync } from "fs";
+import { API_VERSION_LIFECYCLE_STAGES } from "./breaking-change.js";
 
 /**
  * @typedef {import('./spec-model.js').Tag} Tag
  * @typedef {import('./spec-model.js').ToJSONOptions} ToJSONOptions
  */
 
+/**
+ * @typedef {Object} Operation
+ * @property {string} id - The operation ID
+ * @property {string} path - API path
+ * @property {string} httpMethod - HTTP method (GET, POST, etc.)
+ */
+
+/**
+ * @typedef {Operation & {swagger: string}} LegacyOperation
```

Comment:
Can we find a better name for this?  What is a "LegacyOperation"?  Should it instead be a pair of operations, like "before" and "after"?

Reply:
Less than the name itself, I care more about the structure.  Should we use a pair of `Operation` objects, instead of `Operation` and `string`?